rclcpp_components/include/rclcpp_components/component_manager_isolated.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef RCLCPP_COMPONENTS__COMPONENT_MANAGER_ISOLATED_HPP__
+#define RCLCPP_COMPONENTS__COMPONENT_MANAGER_ISOLATED_HPP__
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include <unordered_map>
+
+#include "rclcpp_components/component_manager.hpp"
+
+
```

Comment:
Its up to core maintainers (@ivanpauno ) but I would rather the other containers follow suit with the `ExecutorT` so that you can remove the constructor field. The template for type + executor constructor argument are redundant. You can have the same effect by removing all constructor arguments from the existing `component_manager.hpp` and making that also templated, which constructs its internal executor by instantiating `ExecutorT()`.

Reply:
Also, needs the function documentation inline (see component_manager.hpp, you added there, just add here too)

Reply:
Yes, we're not taking any advantage of the template parameter here.
We can delete it or be consistent in the base class, I would just remove it as it doesn't seem to be needed.

(with the template parameter we could've not added the new virtual methods, but both approaches are fine)